zebedee-cms/src/main/java/com/github/onsdigital/zebedee/service/ZebedeeDatasetService.java

```diff
@@ -28,11 +30,44 @@ public ZebedeeDatasetService(DatasetClient datasetClient) {
         this.datasetClient = datasetClient;
     }
 
+    private ContentStatus updatedStateInCollection(ContentStatus currentState, ContentStatus newState, String lastEditedBy, String user) throws ForbiddenException, BadRequestException {
```

Comment:
I know this is MVP but I would consider checking if each of method params is null before doing anything and throwing the appropriate exception.

Or you could just update the order of values in the equals to make it null safe - just swap the order of the values so you are comparing the constant to the variable. For example
```
if (currentState == null && ContentStatus.Reviewed.equals(newState)) {...}
```
In this case even if `newState` is null it wont throw an null pointer exception.

Also I would consider adding logging to each case (error or a valid state update) - sees like something we want to trace & possible audit. I'll leave that up to you but if there is a bug logging will be super useful...

Reply:
We're already doing a null check before we call this function:
https://github.com/ONSdigital/zebedee/pull/138/files/5c7294f263ca380ea44b722e9078ed69b48ab322#diff-bbf23d72826f8b295a4591a7f223131cL112

However, make sure it never fails, even if it's called elsewhere, I've added a null check into the `updatedStateInCollection` block.

Reply:
I've added some logging now too:
https://github.com/ONSdigital/zebedee/pull/138/commits/edb0f5306d1d58c0a7c0f7a48ce73a03e8140254